bug with darshan test cpi.c

Source: https://github.com/buildtesters/buildtest-cori/blob/devel/buildspecs/apps/darshan/cpi.c
Buildspecs: https://github.com/buildtesters/buildtest-cori/blob/devel/buildspecs/apps/darshan/darshan-hello.yml
[buildtest.log](https://github.com/buildtesters/buildtest-cori/files/5990963/test.log)

```
(buildtest) siddiq90@cori01:~/buildtest-cori/buildspecs> buildtest inspect f41571ca
{
  "id": "f41571ca",
  "full_id": "f41571ca-3e21-4803-b86a-f1f62372eaa3",
  "testroot": "/global/u1/s/siddiq90/buildtest/var/tests/slurm.e4s/darshan-hello/darshan_cpi_example/1",
  "testpath": "/global/u1/s/siddiq90/buildtest/var/tests/slurm.e4s/darshan-hello/darshan_cpi_example/1/stage/generate.sh",
  "command": "sbatch --parsable --clusters=cori --account=nstaff -q regular -C knl -t 10 -n 4 /global/u1/s/siddiq90/buildtest/var/tests/slurm.e4s/darshan-hello/darshan_cpi_example/1/stage/generate.sh",
  "outfile": "/global/u1/s/siddiq90/buildtest/var/tests/slurm.e4s/darshan-hello/darshan_cpi_example/1/stage/darshan_cpi_example.out",
  "errfile": "/global/u1/s/siddiq90/buildtest/var/tests/slurm.e4s/darshan-hello/darshan_cpi_example/1/stage/darshan_cpi_example.err",
  "schemafile": "compiler-v1.0.schema.json",
  "executor": "slurm.e4s",
  "tags": "e4s",
  "starttime": "2021-02-16T10:07:42",
  "endtime": "2021-02-16T10:10:52",
  "runtime": "00:03:10",
  "state": "PASS",
  "returncode": 0,
  "output": "",
  "error": "/usr/bin/ld: /tmp/iccRGlr8H.o: undefined reference to symbol 'MPI_Wtime'\n/usr/bin/ld: /opt/cray/pe/mpt/7.7.10/gni/mpich-intel/16.0/lib/libmpich_intel.so.3: error adding symbols: DSO missing from command line\nsrun: fatal: Can not execute cpi.c.exe\nError: failed to open darshan log file /global/homes/s/siddiq90/mpi_c_basic_offset.darshan.\nrm: cannot remove '/global/homes/s/siddiq90/mpi_c_basic_offset.darshan': No such file or directory\n",
  "job": {
    "Account": "nstaff",
    "AllocNodes": "1",
    "AllocTRES": "billing=272,cpu=272,energy=23174,mem=87G,node=1",
    "ConsumedEnergyRaw": "23174",
    "CPUTimeRaw": "51680",
    "Elapsed": "00:03:10",
    "End": "2021-02-16T10:10:52",
    "ExitCode": "0:0",
    "JobID": "39448912",
    "JobName": "darshan_cpi_example",
    "NCPUS": "272",
    "NNodes": "1",
    "QOS": "regular_1",
    "ReqGRES": "PER_NODE:craynetwork:1",
    "ReqMem": "87Gn",
    "ReqNodes": "1",
    "ReqTRES": "billing=4,cpu=4,node=1",
    "Start": "2021-02-16T10:07:42",
    "State": "COMPLETED",
    "Submit": "2021-02-16T10:06:41",
    "UID": "92503",
    "User": "siddiq90",
    "WorkDir": "/global/u1/s/siddiq90/buildtest/var/tests/slurm.e4s/darshan-hello/darshan_cpi_example/1/stage\n",
    "scontrol": {
      "command": "scontrol show job 39448912 --clusters=cori",
      "output": "JobId=39448912 JobName=darshan_cpi_example\n   UserId=siddiq90(92503) GroupId=siddiq90(92503) MCS_label=N/A\n   Priority=69119 Nice=0 Account=nstaff QOS=regular_1\n   JobState=COMPLETING Reason=None Dependency=(null)\n   Requeue=0 Restarts=0 BatchFlag=1 Reboot=0 ExitCode=0:0\n   RunTime=00:03:10 TimeLimit=00:10:00 TimeMin=N/A\n   SubmitTime=2021-02-16T10:06:41 EligibleTime=2021-02-16T10:06:41\n   AccrueTime=2021-02-16T10:06:50\n   StartTime=2021-02-16T10:07:42 EndTime=2021-02-16T10:10:52 Deadline=N/A\n   PreemptEligibleTime=2021-02-16T10:07:42 PreemptTime=None\n   SuspendTime=None SecsPreSuspend=0 LastSchedEval=2021-02-16T10:07:42\n   Partition=regular_knl AllocNode:Sid=cori01:55352\n   ReqNodeList=(null) ExcNodeList=(null)\n   NodeList=nid04270\n   BatchHost=nid04270\n   NumNodes=1 NumCPUs=272 NumTasks=4 CPUs/Task=1 ReqB:S:C:T=0:0:*:*\n   TRES=cpu=272,mem=87G,energy=23174,node=1,billing=272\n   Socks/Node=* NtasksPerN:B:S:C=0:0:*:* CoreSpec=*\n   MinCPUsNode=1 MinMemoryNode=87G MinTmpDiskNode=0\n   Features=knl&quad&cache DelayBoot=2-00:00:00\n   OverSubscribe=NO Contiguous=0 Licenses=(null) Network=(null)\n   Command=/global/u1/s/siddiq90/buildtest/var/tests/slurm.e4s/darshan-hello/darshan_cpi_example/1/stage/generate.sh\n   WorkDir=/global/u1/s/siddiq90/buildtest/var/tests/slurm.e4s/darshan-hello/darshan_cpi_example/1/stage\n   AdminComment={\"stderrPath\":\"\\/global\\/u1\\/s\\/siddiq90\\/buildtest\\/var\\/tests\\/slurm.e4s\\/darshan-hello\\/darshan_cpi_example\\/1\\/stage\\/darshan_cpi_example.err\",\"partition\":\"regular_knl\",\"stdoutPath\":\"\\/global\\/u1\\/s\\/siddiq90\\/buildtest\\/var\\/tests\\/slurm.e4s\\/darshan-hello\\/darshan_cpi_example\\/1\\/stage\\/darshan_cpi_example.out\",\"jobDerivedExitCode\":0,\"uid\":92503,\"timeLimit\":10,\"tresRequest\":\"1=272,2=89088,3=18446744073709551614,4=1,5=272\",\"reboot\":0,\"jobId\":39448912,\"jobAccount\":\"nstaff\",\"arrayTaskId\":4294967294,\"resizing\":0,\"name\":\"darshan_cpi_example\",\"priority\":69119,\"packJobId\":0,\"nodes\":\"nid04270\",\"stdinPath\":\"\\/dev\\/null\",\"allocNodes\":1,\"restartCnt\":0,\"cluster\":\"cori\",\"submitTime\":1613498801,\"workingDirectory\":\"\\/global\\/u1\\/s\\/siddiq90\\/buildtest\\/var\\/tests\\/slurm.e4s\\/darshan-hello\\/darshan_cpi_example\\/1\\/stage\",\"gresUsed\":\"craynetwork:4\",\"packJobOffset\":0,\"endTime\":1613499052,\"argv\":[\"\\/global\\/u1\\/s\\/siddiq90\\/buildtest\\/var\\/tests\\/slurm.e4s\\/darshan-hello\\/darshan_cpi_example\\/1\\/stage\\/generate.sh\"],\"qos\":\"regular_1\",\"batchHost\":\"nid04270\",\"startTime\":1613498862,\"arrayJobId\":0,\"gresRequest\":\"PER_NODE:craynetwork:1\",\"allocCpus\":272,\"features\":\"knl&quad&cache\",\"jobExitCode\":0} \n   StdErr=/global/u1/s/siddiq90/buildtest/var/tests/slurm.e4s/darshan-hello/darshan_cpi_example/1/stage/darshan_cpi_example.err\n   StdIn=/dev/null\n   StdOut=/global/u1/s/siddiq90/buildtest/var/tests/slurm.e4s/darshan-hello/darshan_cpi_example/1/stage/darshan_cpi_example.out\n   Power=\n   TresPerNode=craynetwork:1\n   MailUser=siddiq90 MailType=NONE\n"
    }
  }
}



Output File
______________________________




Error File
______________________________
/usr/bin/ld: /tmp/iccRGlr8H.o: undefined reference to symbol 'MPI_Wtime'
/usr/bin/ld: /opt/cray/pe/mpt/7.7.10/gni/mpich-intel/16.0/lib/libmpich_intel.so.3: error adding symbols: DSO missing from command line
srun: fatal: Can not execute cpi.c.exe
Error: failed to open darshan log file /global/homes/s/siddiq90/mpi_c_basic_offset.darshan.
rm: cannot remove '/global/homes/s/siddiq90/mpi_c_basic_offset.darshan': No such file or directory




Test Content
______________________________
#!/bin/bash 
#SBATCH --time=10
#SBATCH --nodes=1
#SBATCH --job-name=darshan_cpi_example
#SBATCH --output=darshan_cpi_example.out
#SBATCH --error=darshan_cpi_example.err
source /global/u1/s/siddiq90/buildtest/var/executors/slurm.e4s/before_script.sh
_EXEC=cpi.c.exe
export DARSHAN_LOGFILE=$HOME/mpi_c_basic_offset.darshan
module swap intel intel/19.1.2.254
spack load darshan-runtime
spack load darshan-util

icc -o $_EXEC /global/u1/s/siddiq90/buildtest-cori/buildspecs/apps/darshan/cpi.c -ldarshan
srun $_EXEC
darshan-parser $DARSHAN_LOGFILE
rm $DARSHAN_LOGFILE

source /global/u1/s/siddiq90/buildtest/var/executors/slurm.e4s/after_script.sh



buildspec:  /global/u1/s/siddiq90/buildtest-cori/buildspecs/apps/darshan/darshan-hello.yml
______________________________
version: "1.0"
buildspecs:
  darshan_cpi_example:
    type: compiler
    executor: slurm.e4s
    description: "MPI test to calculate PI and use darshan-parser to view logfile"
    tags: [e4s]
    source: cpi.c
    compilers:
      name: ["^(intel/19.1.2.254)$"]
      default:
        intel:
          ldflags: -ldarshan
          batch:
            timelimit: "10"
            nodecount: "1"
          env: 
            DARSHAN_LOGFILE: "$HOME/mpi_c_basic_offset.darshan"
          pre_build: |
            spack load darshan-runtime
            spack load darshan-util
          run: srun $_EXEC
          post_run: |
            darshan-parser $DARSHAN_LOGFILE
            rm $DARSHAN_LOGFILE
      config:
        intel/19.1.2.254:
          module:
            swap: [intel, intel/19.1.2.254]

maintainers:
  - "Shahzeb Siddiqui <@shahzebsiddiqui>"
```

This is the setup script sourced from executor

```
(buildtest) siddiq90@cori01:~/buildtest-cori/buildspecs> cat [/global/u1/s/siddiq90/buildtest/var/executors/slurm.e4s/before_script.sh](url)
source /global/common/software/spackecp/e4s-20.10/spack/share/spack/setup-env.sh
cd $SCRATCH/testsuite
source setup.sh
```


Output of build

```
(buildtest) siddiq90@cori01:~/buildtest-cori/buildspecs> buildtest build -b apps/darshan/darshan-hello.yml 

+-------------------------------+
| Stage: Discovering Buildspecs |
+-------------------------------+ 
    

Discovered Buildspecs:
 
/global/u1/s/siddiq90/buildtest-cori/buildspecs/apps/darshan/darshan-hello.yml

+---------------------------+
| Stage: Parsing Buildspecs |
+---------------------------+ 
    
 schemafile                | validstate   | buildspec
---------------------------+--------------+--------------------------------------------------------------------------------
 compiler-v1.0.schema.json | True         | /global/u1/s/siddiq90/buildtest-cori/buildspecs/apps/darshan/darshan-hello.yml

+----------------------+
| Stage: Building Test |
+----------------------+ 



 name                | id       | type     | executor   | tags    | compiler         | testpath
---------------------+----------+----------+------------+---------+------------------+-----------------------------------------------------------------------------------------------------------
 darshan_cpi_example | f41571ca | compiler | slurm.e4s  | ['e4s'] | intel/19.1.2.254 | /global/u1/s/siddiq90/buildtest/var/tests/slurm.e4s/darshan-hello/darshan_cpi_example/1/stage/generate.sh

+----------------------+
| Stage: Running Test  |
+----------------------+ 
    
[darshan_cpi_example] JobID: 39448912 dispatched to scheduler
 name                | id       | executor   | status   |   returncode | testpath
---------------------+----------+------------+----------+--------------+-----------------------------------------------------------------------------------------------------------
 darshan_cpi_example | f41571ca | slurm.e4s  | N/A      |           -1 | /global/u1/s/siddiq90/buildtest/var/tests/slurm.e4s/darshan-hello/darshan_cpi_example/1/stage/generate.sh


Polling Jobs in 10 seconds
________________________________________
[darshan_cpi_example]: JobID 39448912 in PENDING state 


Polling Jobs in 10 seconds
________________________________________
[darshan_cpi_example]: JobID 39448912 in PENDING state 


Polling Jobs in 10 seconds
________________________________________
[darshan_cpi_example]: JobID 39448912 in PENDING state 


Polling Jobs in 10 seconds
________________________________________
[darshan_cpi_example]: JobID 39448912 in PENDING state 


Polling Jobs in 10 seconds
________________________________________
[darshan_cpi_example]: JobID 39448912 in PENDING state 


Polling Jobs in 10 seconds
________________________________________
[darshan_cpi_example]: JobID 39448912 in PENDING state 


Polling Jobs in 10 seconds
________________________________________
[darshan_cpi_example]: JobID 39448912 in RUNNING state 


Polling Jobs in 10 seconds
________________________________________
[darshan_cpi_example]: JobID 39448912 in RUNNING state 


Polling Jobs in 10 seconds
________________________________________
[darshan_cpi_example]: JobID 39448912 in RUNNING state 


Polling Jobs in 10 seconds
________________________________________
[darshan_cpi_example]: JobID 39448912 in RUNNING state 


Polling Jobs in 10 seconds
________________________________________
[darshan_cpi_example]: JobID 39448912 in RUNNING state 


Polling Jobs in 10 seconds
________________________________________
[darshan_cpi_example]: JobID 39448912 in RUNNING state 


Polling Jobs in 10 seconds
________________________________________
[darshan_cpi_example]: JobID 39448912 in RUNNING state 


Polling Jobs in 10 seconds
________________________________________
[darshan_cpi_example]: JobID 39448912 in RUNNING state 


Polling Jobs in 10 seconds
________________________________________
[darshan_cpi_example]: JobID 39448912 in RUNNING state 


Polling Jobs in 10 seconds
________________________________________
[darshan_cpi_example]: JobID 39448912 in RUNNING state 


Polling Jobs in 10 seconds
________________________________________
[darshan_cpi_example]: JobID 39448912 in RUNNING state 


Polling Jobs in 10 seconds
________________________________________
[darshan_cpi_example]: JobID 39448912 in RUNNING state 


Polling Jobs in 10 seconds
________________________________________
[darshan_cpi_example]: JobID 39448912 in RUNNING state 


Polling Jobs in 10 seconds
________________________________________
[darshan_cpi_example]: JobID 39448912 in RUNNING state 


Polling Jobs in 10 seconds
________________________________________
[darshan_cpi_example]: JobID 39448912 in RUNNING state 


Polling Jobs in 10 seconds
________________________________________
[darshan_cpi_example]: JobID 39448912 in RUNNING state 


Polling Jobs in 10 seconds
________________________________________
[darshan_cpi_example]: JobID 39448912 in RUNNING state 


Polling Jobs in 10 seconds
________________________________________
[darshan_cpi_example]: JobID 39448912 in RUNNING state 


Polling Jobs in 10 seconds
________________________________________
[darshan_cpi_example]: JobID 39448912 in COMPLETED state 


Polling Jobs in 10 seconds
________________________________________

    +---------------------------------------------+
    | Stage: Final Results after Polling all Jobs |
    +---------------------------------------------+ 
    
 name                | id       | executor   | status   |   returncode | testpath
---------------------+----------+------------+----------+--------------+-----------------------------------------------------------------------------------------------------------
 darshan_cpi_example | f41571ca | slurm.e4s  | PASS     |            0 | /global/u1/s/siddiq90/buildtest/var/tests/slurm.e4s/darshan-hello/darshan_cpi_example/1/stage/generate.sh

+----------------------+
| Stage: Test Summary  |
+----------------------+ 

Executed 1 tests
Passed Tests: 1/1 Percentage: 100.000%
Failed Tests: 0/1 Percentage: 0.000%



Writing Logfile to: /tmp/buildtest_8m6plh1w.log
```

--- a/buildspecs/apps/darshan/darshan-hello.yml
+++ b/buildspecs/apps/darshan/darshan-hello.yml
@@ -10,6 +10,7 @@ buildspecs:
       name: ["^(intel/19.1.2.254)$"]
       default:
         intel:
+          cc: cc
           ldflags: -ldarshan
           batch:
             timelimit: "10"

